index.js

```diff
@@ -3,156 +3,132 @@
+ * Creates a deprecation warning item.
+ * @function
+ * @memberof processWarning
+ * @param {ProcessWarningOptions} params - Options for creating the warning.
+ * @returns {WarningItem} The created deprecation warning item.
  */
-function processWarning () {
-  const codes = {}
-  const emitted = new Map()
-  const opts = Object.create(null)
-
-  /**
-   * Builds a new {@link ProcessWarning} and adds it to the
-   * {@link ProcessWarningManager} such that it can be emitted through the
-   * {@link ProcessWarningManager#emit} method.
-   *
-   * @memberof! ProcessWarningManager
-   * @instance
-   *
-   * @param {string} name Warning name, e.g. `'MyCustomWarning'`.
-   * @param {string} code A unique code for the warning, e.g. `'WARN_001'`.
-   * @param {string} message The body of the warning.
-   * @param {CreateOptions} [options]
-   * @returns {ProcessWarningBuilder}
-   */
-  function create (name, code, message, { unlimited = false } = {}) {
-    if (!name) throw new Error('Warning name must not be empty')
-    if (!code) throw new Error('Warning code must not be empty')
-    if (!message) throw new Error('Warning message must not be empty')
-    if (typeof unlimited !== 'boolean') throw new Error('Warning opts.unlimited must be a boolean')
+function createDeprecation (params) {
+  return createWarning({ ...params, name: 'DeprecationWarning' })
+}
 
-    code = code.toUpperCase()
+/**
+ * Emits the warning.
+ * @typedef WarningEmitter
+ * @param {*} [a] Possible message interpolation value.
+ * @param {*} [b] Possible message interpolation value.
+ * @param {*} [c] Possible message interpolation value.
+ */
 
-    if (codes[code] !== undefined) {
-      throw new Error(`The code '${code}' already exist`)
-    }
+/**
+ * Creates a warning item.
+ * @function
+ * @memberof processWarning
+ * @param {ProcessWarningOptions} params - Options for creating the warning.
+ * @returns {WarningItem} The created warning item.
+ * @throws {Error} Throws an error if name, code, or message is empty, or if opts.unlimited is not a boolean.
+ */
+function createWarning ({ name, code, message, unlimited = false } = {}) {
+  if (!name) throw new Error('Warning name must not be empty')
+  if (!code) throw new Error('Warning code must not be empty')
+  if (!message) throw new Error('Warning message must not be empty')
+  if (typeof unlimited !== 'boolean') throw new Error('Warning opts.unlimited must be a boolean')
 
-    function buildWarnOpts (a, b, c) {
-      // more performant than spread (...) operator
-      let formatted
-      if (a && b && c) {
-        formatted = format(message, a, b, c)
-      } else if (a && b) {
-        formatted = format(message, a, b)
-      } else if (a) {
-        formatted = format(message, a)
-      } else {
-        formatted = message
-      }
+  code = code.toUpperCase()
 
-      return {
-        code,
-        name,
-        message: formatted
-      }
-    }
+  const warningContainer = unlimited
```

Comment:
@jsumners requested to change it to
```js
  let warningContainer = {
    [name]: function (a, b, c) {
      if (warning.emitted === true && warning.unlimited !== true) {
        return
      }
      warning.emitted = true

      process.emitWarning(warning.format(a, b, c), warning.name, warning.code)
    }
  }
  if (unlimited) {
    warningContainer = {
      [name]: function (a, b, c) {
        warning.emitted = true

        process.emitWarning(warning.format(a, b, c), warning.name, warning.code)
      }
    }
  }
```

Could not provide suggestion because empty lines are there